test/view/iota.cpp

```diff
@@ -33,62 +34,36 @@ struct Int
     bool operator!=(Int j) const { return i != j.i; }
 };
 
-template<typename Integral>
-void test_iota_minus() {
+CPP_template(typename I)(
+    requires ranges::Integral<I>)
+void test_iota_distance()
+{
     using namespace ranges;
-    using D = detail::iota_difference_t<Integral>;
-    using I = Integral;
-    Integral max = std::numeric_limits<Integral>::max();
-
-    CHECK(detail::iota_minus_(I(0), I(0)) == D(0));
-    CHECK(detail::iota_minus_(I(0), I(1)) == D(-1));
-    CHECK(detail::iota_minus_(I(1), I(0)) ==  D(1));
-    CHECK(detail::iota_minus_(I(1), I(1)) == D(0));
-
-    CHECK(detail::iota_minus_(I(max - I(1)), I(max - I(1))) == D(0));
-    CHECK(detail::iota_minus_(I(max - I(1)), I(max)) == D(-1));
-    CHECK(detail::iota_minus_(I(max), I(max - I(1))) == D(1));
-    CHECK(detail::iota_minus_(I(max), I(max)) == D(0));
+    using D = iter_difference_t<I>;
+    I max = std::numeric_limits<I>::max();
+
+    CHECK(detail::iota_distance_(I(0), I(0)) == D(0));
+    CHECK(detail::iota_distance_(I(1), I(0)) == D(-1));
+    CHECK(detail::iota_distance_(I(0), I(1)) ==  D(1));
+    CHECK(detail::iota_distance_(I(1), I(1)) == D(0));
+
+    CHECK(detail::iota_distance_(I(max - I(1)), I(max - I(1))) == D(0));
+    CHECK(detail::iota_distance_(I(max), I(max - I(1))) == D(-1));
+    CHECK(detail::iota_distance_(I(max - I(1)), I(max)) == D(1));
+    CHECK(detail::iota_distance_(I(max), I(max)) == D(0));
 }
 
-template<typename U>
-void test_wrap_around() {
-    using namespace ranges;
-    auto max = std::numeric_limits<U>::max();
-    U from = max - 2;
-    U to = 2;
-
-    auto rng = view::ints(from, to);
-    static_assert(RandomAccessRange<decltype(rng)>, "");
-    static_assert(SizedRange<decltype(rng)>, "");
-    models<RandomAccessViewConcept>(aux::copy(rng));
-    models<BoundedViewConcept>(aux::copy(rng));
-    CHECK(ranges::size(rng) == std::size_t(5));
-    ::check_equal(rng, {U(max - 2), U(max - 1), max, U(0), U(1)});
-
-    auto rng2 = view::indices(from, to);
-    static_assert(RandomAccessRange<decltype(rng2)>, "");
-    static_assert(SizedRange<decltype(rng2)>, "");
-    models<RandomAccessViewConcept>(aux::copy(rng2));
-    models<BoundedViewConcept>(aux::copy(rng2));
-    CHECK(ranges::size(rng2) == std::size_t(5));
-    ::check_equal(rng2, {U(max - 2), U(max - 1), max, U(0), U(1)});
-
-    auto rng3 = view::closed_indices(from, to);
-    static_assert(RandomAccessRange<decltype(rng3)>, "");
-    static_assert(SizedRange<decltype(rng3)>, "");
-    models<RandomAccessViewConcept>(aux::copy(rng3));
-    models<BoundedViewConcept>(aux::copy(rng3));
-    CHECK(ranges::size(rng3) == std::size_t(6));
-    ::check_equal(rng3, {U(max - 2), U(max - 1), max, U(0), U(1), U(2)});
-}
+template<class> struct undef;
```

Comment:
Put this in `simple_test.hpp` so we don't have to constantly define it (and clean it out of PRs).